packages/nimble-components/src/banner/styles.ts

```diff
@@ -0,0 +1,214 @@
+    Information100DarkUi,
+    Information100LightUi,
+    Warning100DarkUi,
+    Warning100LightUi,
+    White
+} from '@ni/nimble-tokens/dist/styledictionary/js/tokens';
+
+import {
+    actionRgbPartialColor,
+    applicationBackgroundColor,
+    bodyFont,
+    borderHoverColor,
+    buttonLabelFontColor,
+    controlHeight,
+    controlSlimHeight,
+    fillSelectedColor,
+    iconColor,
+    iconSize,
+    linkActiveFontColor,
+    linkDisabledFontColor,
+    linkFontColor,
+    standardPadding
+} from '../theme-provider/design-tokens';
+import { Theme } from '../theme-provider/types';
+import { hexToRgbaCssColor } from '../utilities/style/colors';
+import { themeBehavior } from '../utilities/style/theme';
+
+export const styles = css`
+    ${display('flex')}
+
+    :host {
+        width: 100%;
+        font: ${bodyFont};
+        font-size: 12.8px;
+        align-items: top;
+        overflow: hidden;
+        color: ${White};
+        ${iconColor.cssCustomProperty}: ${hexToRgbaCssColor(White, 0.6)};
+    }
+
+    :host(:not([open])) {
+        display: none;
+    }
+
+    .container {
+        display: flex;
+        width: 100%;
+    }
+
+    .icon {
+        width: 48px;
+        display: flex;
+        justify-content: center;
+        margin-top: 8px;
+        flex: 0 0 auto;
+    }
+
+    .text {
+        display: inline;
+        margin-top: 7px;
+        margin-bottom: 7px;
+    }
+
+    slot[name='title']::slotted(*) {
+        font-weight: bold;
+        padding-right: 8px;
+        white-space: nowrap;
+    }
+
+    :host([title-hidden]) slot[name='title'] {
+        ${
+            /**
+             * Hide content visually while keeping it screen reader-accessible.
+             * Source: https://webaim.org/techniques/css/invisiblecontent/#techniques
+             * See discussion here: https://github.com/microsoft/fast/issues/5740#issuecomment-1068195035
+             */
+            ''
+        }
+        display: inline-block;
+        height: 1px;
+        width: 1px;
+        position: absolute;
+        margin: -1px;
+        clip: rect(1px, 1px, 1px, 1px);
+        clip-path: inset(50%);
+        overflow: hidden;
+        padding: 0;
+    }
+
+    .controls {
+        height: ${controlHeight};
+        margin-left: auto;
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        align-self: flex-start;
+    }
+
+    slot[name='action'] {
+        display: flex;
+        align-content: center;
+        margin-left: ${standardPadding};
+        white-space: nowrap;
+    }
+
+    slot[name='action']::slotted(nimble-anchor) {
+        ${linkFontColor.cssCustomProperty}: ${White};
+        ${linkDisabledFontColor.cssCustomProperty}: ${White};
+        ${linkActiveFontColor.cssCustomProperty}: ${hexToRgbaCssColor(
+            White,
+            0.6
+        )};
+        font-size: 12.8px;
+    }
+
+    slot[name='action']::slotted(nimble-button) {
+        height: ${controlSlimHeight};
```

Comment:
Our controls are wonky with how you define height, they base their sizing on the actual value of the controlHeight  css property. So we should set that value so it propagates to the control correctly. See: https://github.com/ni/nimble/issues/610

So this should be like the other properties being defined here:

```suggestion
        ${controlHeight.cssCustomProperty}: ${controlSlimHeight};
```

This manifests in icon buttons, see how the width is incorrect:
![image](https://user-images.githubusercontent.com/1588923/218886697-5f1c7f1e-2a79-4750-9863-5376a6978342.png)

Due to relying on [controlHeight](https://github.com/ni/nimble/blob/28a56104a74fa27e5a1e8f492203669ce28e0048/packages/nimble-components/src/patterns/button/styles.ts#L70).

So I do think we should use the controlHeight instead of setting the height due to our current sizing architecture. A separate question is are we supposed to support the action button as an icon button? Then we need it in the theme matrix and storybook as well.

Reply:
I wasn't planning on supporting icon buttons, but with your suggested fix, there's no reason not to. I've added them to our Storybook stories.